File: ZCopyUnitTester/Mocks/MockBuilder_IFileComparer.vb
Imports System.IO
Imports Moq
Imports ZCopy

Partial Public Class MockBuilder

    Public Shared Function CreateIFileComparer(isSameFile As Boolean) As Mock(Of IFileComparer)
        Dim comparer As New Mock(Of IFileComparer)()
        comparer.Setup(Function(x) x.IsSamefile(It.IsAny(Of String), It.IsAny(Of String))).Returns(isSameFile)
        comparer.Setup(Function(x) x.IsSameFile(It.IsAny(Of FileInfo), It.IsAny(Of FileInfo))).Returns(isSameFile)
        Return comparer
    End Function

End Class

File: ZCopyUnitTester/Mocks/MockBuilder_IFileComparer.cs
﻿using System.IO;
using GenericClassLibrary.FileSystem.Compare;
using Moq;

namespace ZCopyUnitTester.Mocks
{
    public partial  class MockBuilder
	{
		public static Mock<IFileComparer> CreateIFileComparer(bool isSameFile )
		{    
			Mock<IFileComparer> comparer = new Mock<IFileComparer>();
			comparer.Setup(x => x.IsSameFile(It.IsAny<string>(), It.IsAny<string>())).Returns(isSameFile);
			comparer.Setup(x => x.IsSameFile(It.IsAny<FileInfo>(), It.IsAny<FileInfo>())).Returns(isSameFile);
			return comparer;
		}
	}
}